// __BEGIN_LICENSE__
//  Copyright (c) 2006-2013, United States Government as represented by the
//  Administrator of the National Aeronautics and Space Administration. All
//  rights reserved.
//
//  The NASA Vision Workbench is licensed under the Apache License,
//  Version 2.0 (the "License"); you may not use this file except in
//  compliance with the License. You may obtain a copy of the License at
//  http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// __END_LICENSE__


package vw.camera;

message TsaiFile {
  optional string name = 1; // Optional Description

  // Intrinsic Properties of the camera
  repeated double focal_length = 2 [packed=true];
  repeated double center_point = 3 [packed=true];
  repeated double u_direction =  4 [packed=true];
  repeated double v_direction =  5 [packed=true];
  repeated double w_direction =  6 [packed=true];

  // Extrinsic Properties
  repeated double camera_center = 7 [packed=true];
  repeated double camera_rotation = 8 [packed=true];

  // Distortion
  required string distortion_name = 9;
  repeated double distortion_vector = 10 [packed=true];

  required double pixel_pitch = 11;
}